File: client/src/services/LinkService/interfaces/Link.ts
import LinkType from './LinkType.ts';

export type YoutubeVideo = {
    videoId: string;
};

type Link = {
    id: string;
    title: string;
    youtube: YoutubeVideo,
    likes: number;
    isLiked: boolean;
    isSaved: boolean;
    belongsToUser: boolean;
    user: string;
    createdAt: string;
    type: LinkType;
    tags: string[];
};

export default Link;

File: server/src/ShareLink.Domain/Models/Link.cs
using ShareLink.Domain.Enums;

namespace ShareLink.Domain.Models;

public class Link
{
    public required string Id { get; init; }

    public required string Title { get; init; }

    public LinkType Type { get; init; }

    public YoutubeData? Youtube { get; init; }

    public ICollection<UserProfile> LikedBy { get; } = new List<UserProfile>();

    public ICollection<UserProfile> SavedBy { get; } = new List<UserProfile>();

    public required string UserId { get; init; }

    public required string UserNickname { get; init; }

    public DateTime CreatedAt { get; init; }

    public required IReadOnlyCollection<Tag> Tags { get; init; }
}
